### PR TITLE
Fixing process documentation

### PR DESCRIPTION
**Pull request by endJunction**
Fixing ogs_file_param and adding stubs for the documentation, some filled.

I'll move the processes in another PR not to mixup the whole thing.
After that you, who are reading this, will have to extend the documentation pages for your processes.

### CODE REVIEW

File: Documentation/ProjectFile/process/t_name.md [hunk 1]
```diff
@@ -1 +1,3 @@
-\ogs_missing_documentation
+Process' name. Reserved for coupled processes.
+
+\todo Extent documentation for coupled processes.
```
Comment (chleh) [inline Documentation/ProjectFile/process/t_name.md]: Extend

File: ProcessLib/HT/CreateHTProcess.cpp [hunk 2]
```diff
@@ -35,7 +35,11 @@ std::unique_ptr<Process> createHTProcess(
     // Process variable.
     auto process_variables = findProcessVariables(
         variables, config,
-        {"temperature", "pressure"});  // configure two Pcs
+        {
+        //! \ogs_file_param_special{process__HT__process_variables__temperature}
+        "temperature",
+        //! \ogs_file_param_special{process__HT__process_variables__pressure}
+        "pressure"});  // configure two Pcs
```
Comment (chleh) [inline ProcessLib/HT/CreateHTProcess.cpp]: Maybe remove `// configure two Pcs`

File: ProcessLib/HT/CreateHTProcess.cpp [hunk 3]
```diff
@@ -135,7 +139,7 @@ std::unique_ptr<Process> createHTProcess(
     // Specific body force parameter.
     Eigen::Vector3d specific_body_force;
     std::vector<double> const b =
-        //! \ogs_file_param_special{process__HT__specific_body_force}
+        //! \ogs_file_param{process__HT__specific_body_force}
         config.getConfigParameter<std::vector<double>>("specific_body_force");
     assert(b.size() > 0 && b.size() < 4);
```
Comment (chleh) [inline ProcessLib/HT/CreateHTProcess.cpp]: That's not docu, but I'd upgrade the assertion to something also checked in release mode.

File: ProcessLib/LIE/HydroMechanics/CreateHydroMechanicsProcess.cpp [hunk 4]
```diff
@@ -47,15 +47,16 @@ std::unique_ptr<Process> createHydroMechanicsProcess(
...
+        //! \ogs_file_param{process__HYDRO_MECHANICS_WITH_LIE__process_variables__process_variable}
+        pv_conf.getConfigParameterList<std::string>("process_variable");
     std::vector<std::reference_wrapper<ProcessVariable>> process_variables;
     for (std::string const& pv_name : range)
     {
-        if (pv_name != "pressure"
-            && pv_name != "displacement"
-            && pv_name.find("displacement_jump")==std::string::npos)
+        if (pv_name != "pressure" && pv_name != "displacement" &&
+            pv_name.find("displacement_jump") == std::string::npos)
```
Comment (chleh) [inline ProcessLib/LIE/HydroMechanics/CreateHydroMechanicsProcess.cpp]: Not docu, but I think the last comparison should be `pv_name.find("displacement_jump") != 0`.

File: ProcessLib/ProcessVariable.h [hunk 5]
```diff
@@ -70,8 +70,20 @@ class ProcessVariable
     std::string const _name;
     MeshLib::Mesh& _mesh;
     const int _n_components;
-    unsigned _shapefunction_order;  ///< Order of the shapefunctions. Requires
-                                    /// appropriate mesh.
+    /// The polynomial order of the process variable's shape functions.
+    ///
+    /// Requires appropriate mesh.
```
Comment (chleh) [inline ProcessLib/ProcessVariable.h]: Requires an...

File: ProcessLib/LIE/HydroMechanics/CreateHydroMechanicsProcess.cpp (hunk 4 above)
Comment (endJunction) [inline ProcessLib/LIE/HydroMechanics/CreateHydroMechanicsProcess.cpp]: Yes. It is already '!=0' in the LIE/SmallDeformation. I fixed that locally and will make a PR later.

**Pull request merged**